packages/edit-site/src/components/page-patterns/use-patterns.js

```diff
@@ -125,32 +133,48 @@ const selectThemePatterns = ( select, { categoryId, search = '' } = {} ) => {
 			hasCategory: ( item ) => ! item.hasOwnProperty( 'categories' ),
 		} );
 	}
-
 	return { patterns, isResolving: false };
 };
 
-const reusableBlockToPattern = ( reusableBlock ) => ( {
-	blocks: parse( reusableBlock.content.raw, {
+const patternBlockToPattern = ( patternBlock, categories ) => ( {
+	blocks: parse( patternBlock.content.raw, {
 		__unstableSkipMigrationLogs: true,
 	} ),
-	categories: reusableBlock.wp_pattern,
-	id: reusableBlock.id,
-	name: reusableBlock.slug,
-	syncStatus: reusableBlock.wp_pattern_sync_status || SYNC_TYPES.full,
-	title: reusableBlock.title.raw,
-	type: reusableBlock.type,
-	reusableBlock,
+	...( patternBlock.wp_pattern_category.length > 0 && {
+		categories: patternBlock.wp_pattern_category.map(
+			( patternCategoryId ) =>
+				categories
+					? categories.find(
+							( category ) => category.id === patternCategoryId
+					  )?.slug
+					: patternCategoryId
+		),
+	} ),
+	id: patternBlock.id,
+	name: patternBlock.slug,
+	syncStatus: patternBlock.wp_pattern_sync_status || SYNC_TYPES.full,
+	title: patternBlock.title.raw,
+	type: 'wp_block',
+	patternBlock,
 } );
 
 const selectUserPatterns = ( select, { search = '', syncStatus } = {} ) => {
 	const { getEntityRecords, getIsResolving } = select( coreStore );
 
 	const query = { per_page: -1 };
 	const records = getEntityRecords( 'postType', USER_PATTERNS, query );
+	const categories = getEntityRecords( 'taxonomy', 'wp_pattern_category', {
+		per_page: -1,
+		_fields: 'id,name,slug',
+		context: 'view',
+	} );
 
 	let patterns = records
-		? records.map( ( record ) => reusableBlockToPattern( record ) )
+		? records.map( ( record ) =>
+				patternBlockToPattern( record, categories )
+		  )
```

Comment:
Could we create a map of the categories for [easier lookup](https://github.com/WordPress/gutenberg/pull/53837/files#diff-0f49c90dc39786ecdf45f374247e433f763d295d80bdaecda3bab11357e4d7fdR147-R149) when converting to a pattern object? Might make a difference when sites have 15K patterns and start to categorise them all.

Reply:
Done.